datastore/db/migrations/0021_update_grants_convenience_fields.py

```diff
@@ -0,0 +1,32 @@
+"""
+The previous migration created new convenience fields on Grant to aid indexing and faster queries,
+This migration follows on to populate the fields on all existing Grants.
+"""
+
+from django.db import migrations
+
+
+UPDATE_GRANTS_SQL = """
+UPDATE db_grant AS g
+SET publisher_org_id = p.org_id,
+    recipient_org_ids = ARRAY(SELECT jsonb_array_elements_text(jsonb_path_query_array((g.data)->'recipientOrganization', '$[*].id'))),
+    funding_org_ids = ARRAY(SELECT jsonb_array_elements_text(jsonb_path_query_array((g.data)->'fundingOrganization', '$[*].id')))
+FROM db_publisher AS p
+WHERE g.publisher_id = p.id;
+"""
```

Comment:
Slightly nervous about running this on 7,606,239 grants.. perhaps we constrain it to `latest`?

Reply:
On a test on my laptop using a snapshot of the database with ~8 million grants, this took less than 5 minutes.